src/rgw/rgw_rest_s3.cc

```diff
@@ -6134,11 +6134,21 @@ const char* RGWSelectObj_ObjStore_S3::header_value_str[3] = {"Records", "applica
 RGWSelectObj_ObjStore_S3::RGWSelectObj_ObjStore_S3():
   s3select_syntax(std::make_unique<s3selectEngine::s3select>()),
   m_s3_csv_object(std::unique_ptr<s3selectEngine::csv_object>()),
+  m_s3_parquet_object(std::unique_ptr<s3selectEngine::parquet_object>()),
```

Comment:
It's not material to merging this PR, but I rather think that having to instantiate a distinct version of the s3select engine for each file format is messier for the consumer than it could be.  Am I correct that we continually re-allocate these, and later, an "rgw_s3select_api" on each request?  I wonder to what extent we could (in future) refactor things so that RGW can re-use something we allocated at startup time (so that the framework allocates just the smallest set of objects needed to process a single query and result).

Reply:
() as for squashing, of course. (I kept the full history for myself)
() as for `rgw_s3select_api`, it reallocates per each request, no need to do that(we can drop the pimpl thing).
() the s3select-engine is reusing objects, we can make the same for RGW(next PR?).

Reply:
all unique_ptr removed
it is allocated on the stack.